Prepare Release 0.5.0 / Update to Substrate 3.0 (#248)
paritytech/substrate-archive: CHANGELOG.md

  - [Changed] no longer need to instantiate a client and manually pass it to the Archive
  - [Changed] rename `run_with` to `drive`
  - [Changed] Archive now accepts just a postgres URL instead of a postgres URL split into its parts. This should
  make configuring the archive more straightforward. Takes from environment variable `DATABASE_URL` if not passed to the
  archive directly
- [Added] Archive now reads the `CHAIN_DATA_DB` environment variable if the path to the backend chain database is not passed directly.
- [Removed] Archive no longer needs an RPC url to function

> - [Changed] Printing out of Postgres URL in plain text

Reviewer: ```suggestion
- [Changed] Print out the Postgres URL at startup
```

…not including the password I hope.

> - [Added] Maximum amount of blocks loaded at once can be configured with `max_block_load`

Reviewer: ```suggestion
- [Added] `max_block_load` to configure the maximum number of blocks loaded at once
```

> - [Added] Two options to configuration file for State Tracing

Reviewer: ```suggestion
- [Added] Two new options to the configuration relative to state tracing:
```

I think we should explain what state tracing actually is somewhere. It's the big feature of this release, so worth spending some text on it imo. (A link to the wiki might suffice here)

Author: Added link to https://github.com/paritytech/substrate-archive/wiki/6.)-State-Tracing-&-Balance-Reconciliation

  - `targets` for specifying runtime targets to trace in WASM
  - `folder` where WASM blobs with tracing enabled are kept.

> - [Changed] Archive config is now separated into sections for readability. Migration can be achieved by looking at the new `archive.conf` in `polkadot-archive` or `node-template-archive` folders.

Reviewer: ```suggestion
- [Changed] Archive config is now separated into sections for readability. Migration is manual but looking at the new `archive.conf` in `polkadot-archive` or `node-template-archive` folders should help.
```

### Internal Changes

> - [Changed] Postgres Queries are now type checked

Reviewer: ```suggestion
- [Changed] Postgres SQL queries are now type checked
```

- [QoL] Refactor file layout to `substrate-archive` and `substrate-archive-backend`.
- [perf] Decouple Database actors
- [QoL] upgrade to SQLx 0.5.0
- [perf] Overhaul of block indexing. Now uses a Iterator to only collect batches of blocks from the database,

> taking advantage of sequential read-speeds. Gathering blocks by RPC is no longer done.

Reviewer: ```suggestion
taking advantage of the better reading performance of sequential data access. Gathering blocks by RPC is no longer done.
```
